trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This file contains builders for some GPflow models supported in Trieste. """
+
+from __future__ import annotations
+
+import math
+from typing import Optional
+
+import gpflow
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflow.models import GPR, SGPR, SVGP, VGP
+
+from ...data import Dataset
+from ...space import SearchSpace
+from ...types import TensorType
+
+
+def build_gpr(
+    data: Dataset,
+    search_space: SearchSpace,
+    kernel_priors: bool = True,
+    trainable_likelihood: bool = False,
+    likelihood_variance: Optional[float] = None,
+) -> GPR:
+    """
+    Constructs a :class:`~gpflow.models.GPR` model with sensible initial parameters and
+    (optionally) priors. We use :class:`~gpflow.kernels.Matern52` kernel in the model as we found
+    it to be effective in most settings. We assume inputs are normalised to the unit hypercube for
+    setting sensible initial parameters for the kernel.
+
+    We set priors for kernel hyperparameters by default in order to stabilize model fitting. We
+    found the priors below to be highly effective for objective functions defined over the unit
+    hypercube. For objective functions with different scaling, other priors will likely be more
+    appropriate. Using priors allows for using maximum a posteriori estimate of these kernel
+    parameters during model fitting.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param trainable_likelihood: If set to `True` Gaussian likelihood paramater is set to
+        non-trainable. By default set to `False`.
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
```

Comment:
I may be wrong but I think if you wanted to have the SNR = 10 you would want to do
```
noise_variance = (0.1 * np.sqrt(empirical_variance))**2
```
I haven't tried that much with automatically setting things like this, but I believe others have:
https://infallible-thompson-49de36.netlify.app/

Reply:
me either, @frgsimpson at some point recommended setting it to a small portion of the empirical variance, but this looks like a reasonable rationale, thanks!

Reply:
np.sqrt(empirical_variance)**2 = empirical_variance, no? 🤔

Reply:
thanks @alansaul  for the link, thats a useful one!
that's correct @frgsimpson, but the other formulation in terms of the SNR is slightly clearer, justifying the parameter a bit better